lib/widgets/actions.dart

```diff
@@ -155,9 +157,9 @@ Future<bool> updateMessageFlagsStartingFromAnchor({
         // No messages were in the range of the request.
         // This should be impossible given that `foundNewest` was false
         // (and that our `numAfter` was positive.)
-        showErrorDialog(context: context,
+        unawaited(showErrorDialog(context: context,
```

Comment:
I just wrote down some thoughts on `showErrorDialog`'s API on a related PR:
https://github.com/zulip/zulip-flutter/pull/937#discussion_r1774377096

so adding a link to that here.

I think the many `unawaited` calls aroung `showErrorDialog` here get fairly noisy. That makes the diff bigger, which is undesirable but could be OK… but then I think it also leaves enough noise in the tree that it might start to undermine the effect of this lint. If we find ourselves scattering `unawaited` calls all over because there are some functions that just need it most times you call them, then the other more interesting `unawaited` calls don't stand out so much.

Probably a good fix for that is to have our own pair of `showFooDialog` wrappers provide an API that's more like the one I describe in that comment.

That would also be a better API for `showDialog` upstream to have. But it's a tough change to make, because the change in return type would break tons of downstream callers; I don't see a clear way to get to a great API given that constraint. So I wouldn't invest in trying to make the change upstream.

Reply:
Yeah, I like the `showSnackBar` API. `Future` is more primitive in the sense that it doesn't carry much contextual information. The wrapper class is a step up from simply using `typedef`, and offers good documentation discoverability.